### PR TITLE
Splitting proto file into 3 distinct files

### PR DESCRIPTION
**Pull request by krucod3**
Issues: #172

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [ ] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

**Comment by krucod3** [comment]
From technical point of view - all done.
What is left to do is taking care of the tracing.

**Comment by krucod3** [comment]
All done now. I also added new requirements for the API lib. Review can be done, but
❗ DON'T MERGE BEFORE v0.3.1 IS ON MAIN ❗ 
We have breaking changes here that must only be released with v0.4.

**Review by lingnoi — CHANGES_REQUESTED** [review]
for the first 37 files, more will follow ...

**Review by lingnoi — COMMENTED** [review]
no more findings

File: grpc/proto/grpc_api.proto [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    map<string, DeleteCondition> dependencies = 2; /// A list of dependencies to other workloads with their corresponding, expected states. Can be used to enable a synchronized stop of a workload.
+}
+
+/**
+* An enum type describing the conditions for deleting a workload. Used for dependency management, and update strategies.
+*/
+enum DeleteCondition {
+    DEL_COND_RUNNING = 0; /// The workload is operational.
+    DEL_COND_NOT_PENDING_NOR_RUNNING = 1; /// The workload is not scheduled or running.
+}
```
[the remark this answers is not in the document]
Comment (krucod3) [reply grpc/proto/grpc_api.proto:114]: I explicitly did not move it as it is only an internal message sent by Ankaios. The user cannot send it and never sees it.

File: doc/docs/usage/upgrading/v0_3_to_v0_4.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+A sample how the new definition of the Control Interface ise used can be found in the examples from [the Ankaios repository](https://github.com/eclipse-ankaios/ankaios).
+
+The reason for splitting some messages into the dedicated file `ank_base.proto`, is that they are also used for the gRPC API of the Ankaios server. This API is mainly used by the Ankaios agents and CLI, but could also be used by third party applications to directly communicate with the Ankaios server. The following chapter details the changes needed to upgrade to v0.4 in case you are using this API.
+
+## gRPC API of the Ankaios Server
+
+Ankaios is designed for an interchangeable communication middleware between the server, the agents and the CLI. The currently available, and thus default, implementation is done with gRPC. In oder to make it more clear which messages belong to the fixed Control Interface and which to the interchangeable gRPC middleware, we have carved out the gRPC API into a separate file named `grpc_api.proto`.
```
[the remark this answers is not in the document]
Comment (krucod3) [reply doc/docs/usage/upgrading/v0_3_to_v0_4.md:22]: Well it sounds better, but:
* adaptable does not mean interchangeable 
* when we say "segregating" we should name both parts or say segregating the gRPC from the Control Interface.

I'll think about a better formulation and add a proposal.

File: common/src/to_server_interface.rs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2023 Elektrobit Automotive GmbH
+// Copyright (c) 2023 Elektrobit Automotive GmbHc
```
[the remark this answers is not in the document]
Comment (krucod3) [reply common/src/to_server_interface.rs:1]: I wonder how did this happen...

File: api/doc/drawio/unit_overview.drawio.svg
[the remark this answers is not in the document]
Comment (krucod3) [reply api/doc/drawio/unit_overview.drawio.svg:1]: You are absolutely right 👍 
Thx!

File: doc/docs/usage/upgrading/v0_3_to_v0_4.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (krucod3) [reply doc/docs/usage/upgrading/v0_3_to_v0_4.md:22]:
```suggestion
Ankaios is designed for an interchangeable communication middleware between the server, the agents and the CLI. The currently available, and thus default, implementation is done with gRPC. In oder to make it more clear which messages belong to the fixed Control Interface and which to the interchangeable gRPC middleware, we have carved out the gRPC API into a separate file named `grpc_api.proto`.



Ankaios facilitates server-agent-CLI communication through an interchangeable middleware, currently implemented using gRPC. By segregating the gRPC API into a distinct `grpc_api.proto` file, we clearly show the target and purpose of this interface.
```

**Comment by krucod3** [comment]
All done or answered.

**Review by lingnoi — APPROVED** [review]
LGTM 👍

**Comment by inf17101** [comment]
I reviewed commit https://github.com/eclipse-ankaios/ankaios/pull/259/commits/2a545a7afeb27a42274e85697020aaea2667112b.

LGTM 👍

**Pull request merged**